server/routes/user.py

```diff
@@ -1,22 +1,99 @@
-from server import app
+from flask import jsonify, request, make_response, current_app
+from server import app, bcrypt, db
+from server.models.User import User
 from textwrap import dedent
+import datetime, uuid, jwt
 
+# Just for development purpose i.e. to display all users in the database
+@app.route("/users/display", methods=["GET"])
+def display():
+    users = User.query.all()
+    output = []
 
+    for user in users:
+        user_data = {}
+        user_data["id"] = user.id
+        user_data["username"] = user.username
+        user_data["email"] = user.email
+        output.append(user_data)
+
+    return jsonify({"users": output})
+
+
+# Login Endpoint
 @app.route("/users/login", methods=["GET", "POST"])
 def login():
-    return dedent(
-        """
-        <h1>USER LOGIN API</h1>
-        <a href="/">Back to home</a>
-    """
+    body = request.json
+    # return jsonify({"user": auth})
+
+    if not body:
+        return make_response(
+            "Could not verify",
+            401,
+            {"WWW-Authenticate": 'Basic realm="Login required!"'},
+        )
+
+    user = User.query.filter_by(email=body["email"]).first()
+
+    if not user:
+        return make_response(
+            "Could not verify",
+            401,
+            {"WWW-Authenticate": 'Basic realm="Login required!"'},
+        )
+
+    if bcrypt.check_password_hash(user.password, body["password"]):
+        token = jwt.encode(
+            {
+                "id": user.id,
+                "exp": datetime.datetime.utcnow() + datetime.timedelta(minutes=30),
+            },
+            app.config["SECRET_KEY"],
+        )
+        return jsonify({"token": token})
+
+    return make_response(
+        "Could not verify", 401, {"WWW-Authenticate": 'Basic realm="Login required!"'}
     )
 
 
-@app.route("/users/signup", methods=["GET", "POST"])
+@app.route("/users/register", methods=["POST"])
 def signup():
-    return dedent(
-        """
-        <h1>USER SIGNUP API</h1>
-        <a href="/">Back to home</a>
     """
-    )
+    This route expected a JSON request body on the format:
```

Comment:
I didn't notice the typo in my previous suggestion :facepalm: It was to be "expects", not "expected" :P 
```suggestion
    This route expectes a JSON request body on the format:
```